pkg/kv/kvserver/store_raft.go

```diff
@@ -320,6 +321,54 @@ func (s *Store) HandleRaftUncoalescedRequest(
 	return enqueue
 }
 
+// checkAndUpdateCrossLocalityRaftMetrics updates store metrics associated with
```

Comment:
Some of my comments on the corresponding `*DistSender` PR apply here too. It's great that we're passing in only what's truly needed 👍🏽 but `checkAndUpdateCrossLocalityRaftMetrics` plus the fact that the method is basically a giant conditional based on `isSent` is a little odd - let's just make two methods, one for receiving and one for sending a message - `updateRaftMetricOn{Incoming,Outgoing}Msg`.

Bonus points for not anchoring on `*Store` - you'll be able to unit test it much simpler if all it needs is a `StoreMetrics` (and takes in the LocalityComparisonType).

Reply:
Done. Used updateRaftMetricOn{Incoming,Outgoing}Msg and anchored the helper function on `StoreMetrics`.